test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go

```diff
@@ -210,6 +210,95 @@ data:
 	})
 }
 
+func Test_SecretsAndConfigMapsWithCustomPathsCanReconcile(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	kubectl := e2e.Kubectl{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+
+	name := "inline-pathsfrom-directorypath"
+	appYaml := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: simple-app2
+  namespace: kappctrl-test
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      paths:
+        file-from-fetch-paths: ""
+        something/blah.txt: ""
+      pathsFrom:
+        - secretRef:
+            name: from-secret
+            directoryPath: dir
+  template:
+  - ytt:
+      inline:
+        pathsFrom:
+        - secretRef:
+            name: from-secret
+            directoryPath: something
+        paths:
+          config.yml: |
+            #@ load("@ytt:data", "data")
+            #@ load("@ytt:yaml", "yaml")
+
+            #@ print(data.list())
+
+            kind: ConfigMap
+            apiVersion: v1
+            metadata:
+              name: foo
+            data:
+              data: #@ yaml.encode(data.list())
+  deploy:
+  - kapp: {}
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: from-secret
+  namespace: kappctrl-test
+stringData:
+  file-from-secret: "thisIsData"
+` + sas.ForNamespaceYAML()
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, e2e.RunOpts{StdinReader: strings.NewReader(appYaml)})
+	})
+
+	logger.Section("check configmap is populated", func() {
+		retry(t, 10*time.Second, func() error {
```

Comment:
Do we need this retry logic? My thought is the kapp deploy should wait for the App to successfully reconcile but did you see the test was not succeeding in some cases?

Reply:
I think the reason it's in the other test in this file is because the secret/configmap triggers the reconciliation of the App (i.e. the secret/configmap for the App is updated causing the App to reconcile). In the test you are writing, the App reconciling should I think be enough to know the configmap exists.